public/controllers/RegisterController.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Descriere:
+ * Author: 
+ */
+
+var app = angular.module('contacts');
+app.controller('RegisterController', function($http, $state) {
+    var vm = this;
+
+    vm.user = {};
+
+    vm.days = [];
+    for (var i = 1; i < 32; i++) {
+        vm.days.push(i);
+    }
+    vm.years = [];
+    for(var i = 1900; i < 2017; i++) {
+        vm.years.push(i);
+    }
+    
+    $http.get('https://restcountries.eu/rest/v1/all').then(function (response) {
+      
+       vm.country = response.data;
+        
+      });
+    vm.checkPasswordValidity = function () {
+        vm.errorMessage = '';
+        if (vm.user.password !== vm.user.confirmPassword) {
+            vm.errorMessage = "Passwords don't match";
+        }
+       
+    };
+
+    vm.register = function () {
+
+        vm.errorMessage = '';
+
+        if (!vm.user.email) {
+            vm.errorMessage = 'Email is required';
+            return;
+        }
+        vm.errorMessage = '';
+       if(!vm.submit){
+        vm.errorMessage = "You don't checked";
```

Comment:
Aici nu ai pus `return;` ceea ce inseamna ca daca user-ul nu check-uieste checkbox-ul, o sa-i dea eroare dar functia o sa ruleze in continuare si o sa trimita request la server oricum. Nu-i bine. Si un mesaj mai frumos ar fi ok :)